eos/form-factors/analytic-d-to-pi.cc

```diff
@@ -0,0 +1,2174 @@
+                            + (378.0 * u2 - 567.0 * u + 189.0) * omega4pi
+                        ) * deltapipi
+                    );
+            };
+            auto I4barI = [&] (const double & u) -> double
+            {
+                const double u2 = u * u;
+                const double ubar = 1.0 - u, ubar2 = ubar * ubar;
+
+                return 1.0 / 96.0 * u2 * ubar2 * (
+                        mpi2 * (
+                            9.0 * (3.0 + 2.0 * ubar * u)
+                            + 9.0 * a2pi * (32.0 * u2 - 26.0 * u - 3.0)
+                        )
+                        + 40.0 * u * (4.0 + 63.0 * ubar * omega4pi) * deltapipi
+                    );
+            };
+            auto I4bar_d1 = [&] (const double & u) -> double
+            {
+                const double u2 = u * u, u3 = u2 * u, u4 = u2 * u2;
+
+                return 1.0 / 48.0 * (
+                        27.0 * mpi2 * (
+                            (10.0 * u4 - 20.0 * u3 + 6.0 * u2 + 4.0 * u - 1.0)
+                            - a2pi * (160.0 * u4 - 300.0 * u3 + 162.0 * u2 - 20.0 * u - 1.0)
+                        )
+                        + 40.0 * u * (
+                            (-40.0 * u2 + 48.0 * u - 12.0)
+                            + 189.0 * (5.0 * u3 - 10.0 *  u2 + 6.0 * u - 1.0) * omega4pi
+                        ) * deltapipi
+                    );
+            };
+            std::function<double (const double &)> integrand(
+                [&] (const double & u) -> double
+                {
+                    const double u2 = u * u;
+
+                    const double tw4psi = pi.psi4(u, mu) - (2.0 * u * mpi2) / (mc2 - q2 + u2 * mpi2) * pi.psi4_i(u, mu);
+                    const double tw4I4bar = (- I4bar_d1(u) + (6.0 * u * mpi2) / (mc2 - q2 + u2 * mpi2) * I4bar(u) + (12.0 * u2 * mpi4) / power_of<2>(mc2 - q2 + u2 * mpi2) * I4barI(u)) * 2.0 * u * mpi2 / (mc2 - q2 + u2 * mpi2);
+
+                    // _select_weight:
+                    //  0.0 -> regular integral
+                    //  1.0 -> integral of derivative w.r.t. -1/M^2
+                    const double weight = (1.0 - _select_weight) + _select_weight * (mc2 - q2 * (1.0 - u) + mpi2 * u * (1.0 - u)) / u;
+
+                    return std::exp(-(mc2 - q2 * (1.0 - u) + mpi2 * u * (1.0 - u)) / (u * _M2)) * weight
+                            * (tw4psi + tw4I4bar) / (mc2 - q2 + u2 * mpi2);
+                }
+            );
+
+            return mc2 * fpi * integrate<GSL::QAGS>(integrand, u0, 1 - 1e-10, config);
+        }
+
+        double Ftil_nlo_tw2(const double & q2, const double & _M2, const double & _select_weight = 0.0) const
+        {
+            // Reminder: q2 is the kinematic variable associated with the momentum
+            // transfer, while s is the kinematic variable in which the function is
+            // analytically continued. See also comment at beginning of Appendix B
+            // of [KMMO2009], p. 21.
```

Comment:
I think that this comment is incorrect, the reference was probably changed, can this be fixed?